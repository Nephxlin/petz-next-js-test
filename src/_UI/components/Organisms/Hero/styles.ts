import styled from "styled-components";

export const HeroContainer = styled.div`
  width: 100%;
  height: 81vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: url('/images/pokemon-hero.jpg') no-repeat  center;
  background-size: cover;

`

export const HeroTitle = styled.h1`
  color: #FFFFFF;
  width: 509px;
  text-align: center;
`