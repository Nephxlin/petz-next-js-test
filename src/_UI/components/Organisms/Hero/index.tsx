import * as S from './styles'

const Hero = () => {

  return (
    <S.HeroContainer data-testid='hero-component'>
      <S.HeroTitle>Cuidamos bem do seu pokémon, para ele cuidar bem de você</S.HeroTitle>
    </S.HeroContainer>
  )
}

export default Hero