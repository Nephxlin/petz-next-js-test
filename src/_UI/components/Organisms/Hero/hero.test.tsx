import 'jest-styled-components';
import { screen, render } from '@testing-library/react';
import Hero from './index';

describe('Hero component', () => {
  it('renders Hero', () => {
    render(
      <Hero />
    );
    const hero = screen.getByTestId('hero-component')
    expect(hero).toBeInTheDocument();
    expect(hero).toHaveStyleRule('background', `url('/images/pokemon-hero.jpg') no-repeat center`);
    expect(hero).toMatchSnapshot();
  });
});
