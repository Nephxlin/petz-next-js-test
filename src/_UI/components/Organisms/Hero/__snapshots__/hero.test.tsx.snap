// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders Hero 1`] = `
.c0 {
  width: 100%;
  height: 81vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: url('/images/pokemon-hero.jpg') no-repeat center;
  background-size: cover;
}

.c1 {
  color: #FFFFFF;
  width: 509px;
  text-align: center;
}

<div
  class="c0"
  data-testid="hero-component"
>
  <h1
    class="c1"
  >
    Cuidamos bem do seu pokémon, para ele cuidar bem de você
  </h1>
</div>
`;
